web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/forvaltning/stonadsstatistikk/StønadsstatistikkMigreringTask.java

```diff
@@ -0,0 +1,122 @@
+import no.nav.foreldrepenger.behandlingslager.fagsak.FagsakProsesstaskRekkefølge;
+import no.nav.foreldrepenger.datavarehus.v2.StønadsstatistikkTjeneste;
+import no.nav.foreldrepenger.skjæringstidspunkt.SkjæringstidspunktTjeneste;
+import no.nav.vedtak.felles.prosesstask.api.ProsessTask;
+import no.nav.vedtak.felles.prosesstask.api.ProsessTaskData;
+import no.nav.vedtak.felles.prosesstask.api.ProsessTaskHandler;
+import no.nav.vedtak.felles.prosesstask.api.ProsessTaskTjeneste;
+
+@Dependent
+@ProsessTask(value = "stønadsstatisikk.migrering", maxFailedRuns = 1)
+@FagsakProsesstaskRekkefølge(gruppeSekvens = false)
+class StønadsstatistikkMigreringTask implements ProsessTaskHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StønadsstatistikkMigreringTask.class);
+    static final String DATO_KEY = "dato";
+    static final String SISTE_DATO = "sistedato";
+    static final String SEKUNDER_PAUSE = "sekunderPause";
+
+    private final BehandlingRepository behandlingRepository;
+    private final EntityManager entityManager;
+    private final StønadsstatistikkTjeneste stønadsstatistikkTjeneste;
+    private final SkjæringstidspunktTjeneste skjæringstidspunktTjeneste;
+    private final BehandlingVedtakRepository behandlingVedtakRepository;
+    private final ProsessTaskTjeneste prosessTaskTjeneste;
+
+    @Inject
+    public StønadsstatistikkMigreringTask(BehandlingRepository behandlingRepository,
+                                          EntityManager entityManager,
+                                          StønadsstatistikkTjeneste stønadsstatistikkTjeneste,
+                                          SkjæringstidspunktTjeneste skjæringstidspunktTjeneste,
+                                          BehandlingVedtakRepository behandlingVedtakRepository,
+                                          ProsessTaskTjeneste prosessTaskTjeneste) {
+        this.behandlingRepository = behandlingRepository;
+        this.entityManager = entityManager;
+        this.stønadsstatistikkTjeneste = stønadsstatistikkTjeneste;
+        this.skjæringstidspunktTjeneste = skjæringstidspunktTjeneste;
+        this.behandlingVedtakRepository = behandlingVedtakRepository;
+        this.prosessTaskTjeneste = prosessTaskTjeneste;
+    }
+
+    @Override
+    public void doTask(ProsessTaskData prosessTaskData) {
+        var dato = LocalDate.parse(prosessTaskData.getPropertyValue(DATO_KEY), DateTimeFormatter.ISO_LOCAL_DATE);
+
+        var saker = finnSakerOpprettetPåDato(dato);
+        LOG.info("Produserer stønadsstatistikk for saker opprettet {} {}", dato, saker.size());
+        saker.forEach(this::produser);
+        opprettNesteTask(prosessTaskData, dato);
+    }
+
+    private void opprettNesteTask(ProsessTaskData prosessTaskData, LocalDate dato) {
+        var sisteDato = LocalDate.parse(prosessTaskData.getPropertyValue(SISTE_DATO), DateTimeFormatter.ISO_LOCAL_DATE);
+        if (dato.isBefore(sisteDato)) {
```

Comment:
er det meningen at migrering skal kjøres for alle datoer til siste dato?

Reply:
ja, tar inn fom og tom i resttjenesten

Reply:
ja, tom er jo gjerne til og med. Men vil det kjøres migrering for sisteDato her?

Reply:
Ja skal kjøres for sistedato også. dato.plusday(1)